subprojects/core/src/main/java/org/gradle/api/internal/plugins/DefaultPluginCollection.java

```diff
@@ -24,12 +24,14 @@
 import org.gradle.api.specs.Spec;
 import org.gradle.api.specs.Specs;
 
-public class DefaultPluginCollection<T extends Plugin> extends DefaultDomainObjectSet<T> implements PluginCollection<T> {
+import java.util.Collection;
 
-    public DefaultPluginCollection(Class<T> type) {
-        super(type);
+class DefaultPluginCollection<T extends Plugin> extends DefaultDomainObjectSet<T> implements PluginCollection<T> {
+
+    public DefaultPluginCollection(Class<T> type, Collection<T> plugins) {
```

Comment:
If the class is now _package private_, both constructors should also be _package private_.